pennylane/templates/repeat.py

```diff
@@ -0,0 +1,211 @@
+
+        We wish to repeat this circuit two times, with each layer having different ``params``:
+
+        .. code-block:: python3
+
+            @qml.qnode(dev)
+            def ansatz(params):
+                qml.repeat(circuit, 2, params)
+                return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))]
+
+        Since we only have one dynamic argument, ``params``, we pass an array of first-dimension two,
+        for the two layers of the ansatz. By looking at ``circuit``, we can see that the ``params`` argument
+        supplies three different parameters to three different gates. Thus, we supply an array of size
+        (2, 3) as an argument to ``qml.repeat``:
+
+        .. code-block:: python3
+
+            params = np.array([[0.5, 0.5, 0.5], [0.4, 0.4, 0.4]])
+
+        which yields the following circuit:
+
+        >>> ansatz(params)
+        >>> print(ansatz.draw())
+        0: ──RX(0.5)──╭RZ(0.5)──RX(0.4)──╭RZ(0.4)──RX(0.3)──╭RZ(0.3)───────────┤ ⟨Z⟩
+        1: ───────────╰RZ(0.5)──RY(0.5)──╰RZ(0.4)──RY(0.4)──╰RZ(0.3)──RY(0.3)──┤ ⟨Z⟩
+
+        **Passing Multiple Static and Dynamic Arguments**
+
+        It is also possible to pass multiple static and dynamic arguments into the same circuit. Dynamic
+        arguments must be ordered in ``qml.repeat`` in the same order in which they are passed into the
+        ``circuit``.
+
+        Consider the following circuit:
+
+        .. code-block:: python3
+
+            def circuit(param1, param2, wires, var):
+                qml.RX(param1, wires=wires[0])
+                qml.MultiRZ(param2, wires=wires)
+
+                if var:
+                    qml.Hadamard(wires=wires[1])
+
+        This circuit can be repeated as:
+
+        .. code-block:: python3
+
+            @qml.qnode(dev)
+            def ansatz(param1, param2):
+                qml.repeat(circuit, 2, param1, param2, wires=[1, 2], var=True)
+                return [qml.expval(qml.PauliZ(1)), qml.expval(qml.PauliZ(2))]
+
+        We can then run the circuit with a given set of parameters (note that the parameters are
+        of size (2, 1), as the circuit is repeated twice, and for each repetition, both ``param1`` and
+        ``param2`` are simply real numbers):
+
+        .. code-block:: python3
+
+            param1 = np.array([0.1, 0.2])
+            param2 = np.array([0.3, 0.4])
+
+        This gives us the following circuit:
+
+        >>> ansatz(parm1, param2)
+        >>> print(ansatz.draw())
+        1: ──RX(0.1)──╭RZ(0.3)──RX(0.2)──╭RZ(0.4)─────┤ ⟨Z⟩
+        2: ───────────╰RZ(0.3)──H────────╰RZ(0.4)──H──┤ ⟨Z⟩
+    """
+
+    ##############
+    # Input checks
+
+    if not isinstance(depth, int):
+        raise ValueError("'depth' must be of type int, got {}".format(type(depth).__name__))
+
+    for arg in args:
+        if len(arg) != depth:
+            raise ValueError(
+                "Each argument in args must have length matching 'depth'; expected {} got {}".format(
```

Comment:
Since the user does not see what "args" is, would it be better to say "each positional argument must..."?